easybuild/easyblocks/x/xmipp.py

```diff
@@ -28,171 +28,223 @@
-            'protocols',
-            os.path.join('libraries', 'bindings', 'python'),
+        deps = [
+            # Name of dependency, name of env var to set, required or not
+            ('HDF5', None, True),
+            ('SQLite', None, True),
+            ('LibTIFF', None, True),
+            ('libjpeg-turbo', None, True),
+            ('Java', 'JAVA_HOME', True),
+            ('MATLAB', 'MATLAB_DIR', False),
         ]
 
-        python_root = get_software_root('Python')
-        if python_root:
-            # extend $PYTHONPATH
-            all_pythonpaths = [os.path.join(self.installdir, p) for p in self.xmipp_pythonpaths]
-            # required so packages installed as extensions in Pythpn dep are picked up
-            all_pythonpaths.append(os.path.join(python_root, pylibdir))
-            all_pythonpaths.append(os.environ.get('PYTHONPATH', ''))
-
-            env.setvar('PYTHONPATH', os.pathsep.join(all_pythonpaths))
+        cuda_root = get_software_root('CUDA')
+        if cuda_root:
+            params.update({'CUDA_BIN': os.path.join(cuda_root, 'bin')})
+            params.update({'CUDA_LIB': os.path.join(cuda_root, 'lib64')})
+            params.update({'NVCC': os.environ['CUDA_CXX']})
+            # Their default for NVCC is to use g++-5, fix that
+            nvcc_flags = '--x cu -D_FORCE_INLINES -Xcompiler -fPIC '
+            nvcc_flags = nvcc_flags + '-Wno-deprecated-gpu-targets -std=c++11'
+            params.update({'NVCC_CXXFLAGS': nvcc_flags})
+            self.use_cuda = True
+
+        for dep in ['CUDA', 'MATLAB']:
+            use_dep = bool(get_software_root(dep))
+            params.update({dep: use_dep})
+
+        if get_software_root('OpenCV'):
+            params.update({'OPENCV': True})
+            if self.use_cuda:
+                params.update({'OPENCVSUPPORTSCUDA': True})
+            if LooseVersion(get_software_version('OpenCV')) >= LooseVersion('3'):
+                params.update({'OPENCV3': True})
+
+        missing_deps = []
+        for dep, var, required in deps:
+            root = get_software_root(dep)
+            if root:
+                if var:
+                    params.update({var: root})
+            elif required:
+                missing_deps.append(dep)
+
+        if missing_deps:
+            raise EasyBuildError("One or more required dependencies not available: %s", ', '.join(missing_deps))
+
+        # Here we know that Java exists since it is a required dependency
+        jnipath = os.path.join(get_software_root('Java'), 'include')
+        jnipath = os.pathsep.join([jnipath, os.path.join(jnipath, 'linux')])
+        params.update({'JNI_CPPPATH': jnipath})
+
+        regex_subs = []
+
+        # Set the variables in the config file to match the environment from EasyBuild
+        for (key, val) in params.items():
+            regex_subs.extend([(r'^%s\s*=.*$' % key, r'%s = %s' % (key, val))])
+
+        apply_regex_substitutions(self.cfgfile, regex_subs)
```

Comment:
Testing this updated easyblock with #9257 lead to:

```
== FAILED: Installation ended unsuccessfully (build directory: /tmp/easybuild_build/Xmipp/3.19.04-Apollo/foss-2019a-Python-2.7.15): build failed (first 300 chars): Failed to patch /tmp/easybuild_build/Xmipp/3.19.04-Apollo/foss-2019a-Python-2.7.15/xmipp.conf: [Errno 2] No such file or directory: '/tmp/easybuild_build/Xmipp/3.19.04-Apollo/foss-2019a-Python-2.7.15/xmipp.conf' -> '/tmp/easybuild_build/Xmipp/3.19.04-Apollo/foss-2019a-Pyth (took 14 sec)
```

Is there supposed to be an existing `xmipp.conf` which you're patching? I coulnd't find a file like that in the build dir.

If you want to create a new file named `xmipp.conf`, you can't use `apply_regex_substitutions`...